src/Rules/Classes/ClassConstantRule.php

```diff
@@ -60,6 +61,24 @@ public function processNode(Node $node, Scope $scope): array
 				$name = $constantString->getValue();
 				$constantNameScopes[$name] = $scope->filterByTruthyValue(new Identical($node->name, new String_($name)));
 			}
+
+			if ($this->checkNonStringableDynamicAccess) {
+				$accepts = $this->ruleLevelHelper->accepts(new StringType(), $nameType, true);
+				$typeResult = $this->ruleLevelHelper->findTypeToCheck(
+					$scope,
+					$node->name,
+					'',
+					static fn (Type $type) => $type->toString()->isString()->yes()
+				);
+
+				if (! $typeResult->getType()->isString()->yes() ||
+					$typeResult->getType()->toString()->isNumericString()->yes()
+				) {
+					$errors[] = RuleErrorBuilder::message(sprintf('Cannot fetch class constant with a non-stringable type %s.', $nameType->describe(VerbosityLevel::typeOnly())))
+						->identifier('classConstant.fetchInvalidExpression')
```

Comment:
The error message and the identifier should reflect we're trying to access a class constant by non-stringable name. That's not obvious.

Reply:
Also it might be nice to mention the class name here.

Reply:
The error message is wrong regarding the isString/toString point you made above. This is no longer about "stringables", but about "strings". Also the identifier should reflect it's about "name".